libvirt/uri/connection_uri.go

```diff
@@ -31,17 +31,16 @@ func Parse(uriStr string) (*ConnectionURI, error) {
 // The name passed to the remote virConnectOpen function is formed by removing
 // transport, hostname, port number, username and extra parameters from the remote URI
 // unless the name option is specified.
-func (u *ConnectionURI) RemoteName() string {
+func (u ConnectionURI) RemoteName() string {
 	q := u.Query()
 	if name := q.Get("name"); name != "" {
 		return name
 	}
 
-	newURI := *u
-	newURI.Scheme = u.driver()
-	newURI.User = nil
-	newURI.Host = ""
-	newURI.RawQuery = ""
+	newURI := url.URL{
+		Scheme: u.driver(),
+		Path:   u.Path,
+	}
```

Comment:
```suggestion
    newURI := *u.URL
	newURI.Scheme = u.driver()
	newURI.User = nil
	newURI.Host = ""
	newURI.RawQuery = ""
```

Reply:
@X-Cli This should be a proper shallow copy of the `url.URL`, could you see if that also fixes the issue for you?

Reply:
@michaelbeaumont This suggestion works too. My feeling is that building a new clean instance, defining only the required fields is less likely to lead to surprises than copying all data then setting the false value on the unnecessary fields. It is also more concise. But ultimately, both suggestions work :)

Reply:
I would generally agree _but_ [the "spec" explicitly defines the remote name by what is removed from the connection URI](https://libvirt.org/uri.html#transport-configuration)

Reply:
Fair point. Thank you for your comments!